### PR TITLE
Automatically convert T_FLOAT ticket count

### PR DESCRIPTION
**Pull request by csfrancis**
@Sirupsen, @byroot 

This will automatically convert floating point ticket counts to integer values instead of raising `TypeError`.

### CODE REVIEW

File: test/resource_test.rb [hunk 1]
```diff
@@ -26,6 +26,12 @@ def test_initialize_invalid_args
     end
   end
 
+  def test_initialize_with_float
+    r = create_resource :testing, tickets: 1.0
+    assert r
+    assert r.tickets == 1
```
Comment (fw42) [inline test/resource_test.rb]: `assert_equal`?

**Comment by fw42** [comment]
:+1:

File: test/resource_test.rb [hunk 2]
```diff
@@ -26,6 +26,12 @@ def test_initialize_invalid_args
     end
   end
 
+  def test_initialize_with_float
+    r = create_resource :testing, tickets: 1.0
+    assert r
```
Comment (byroot) [inline test/resource_test.rb]: single letter variables makes me :crying_cat_face:

**Comment by byroot** [comment]
Agree with Flo on the `assert_equal`.

Other than that :+1:

**Pull request merged**